autoarray/plot/mat_wrap/wrap/wrap_base.py

```diff
@@ -0,0 +1,946 @@
+            relevent for the projects PyAutoGalaxy / PyAutoLens).
+        """
+
+        self.use_scaled = use_scaled
+        self.conversion_factor = conversion_factor
+        self.in_kpc = in_kpc
+
+        if use_scaled is not None:
+            self.use_scaled = use_scaled
+        else:
+            try:
+                self.use_scaled = conf.instance["visualize"]["general"]["general"][
+                    "use_scaled"
+                ]
+            except:
+                self.use_scaled = True
+
+        try:
+            self.in_kpc = (
+                in_kpc
+                if in_kpc is not None
+                else conf.instance["visualize"]["general"]["units"]["in_kpc"]
+            )
+        except:
+            self.in_kpc = None
+
+
+class AbstractMatWrap:
+    def __init__(self, kwargs: dict):
+        """
+        An abstract base class for wrapping matplotlib plotting methods.
+        
+        Classes are used to wrap matplotlib so that the data structures in the `autoarray.structures` package can be 
+        plotted in standardized withs. This exploits how these structures have specific formats, units, properties etc.
+        This allows us to make a simple API for plotting structures, for example to plot an `Array` structure:
+        
+        import autoarray as aa
+        import autoarray.plot as aplt
+        
+        arr = aa.Array.manual_2d(array=[[1.0, 1.0], [2.0, 2.0]], pixel_scales=2.0)
+        aplt.Array(array=arr)
+        
+        The wrapped Mat objects make it simple to customize how matplotlib visualizes this data structure, for example
+        we can customize the figure size and colormap using the `Figure` and `Cmap` objects.
+        
+        figure = aplt.Figure(figsize=(7,7), aspect="square")
+        cmap = aplt.Cmap(cmap="jet", vmin=1.0, vmax=2.0)
+        
+        plotter = aplt.MatPlot2D(figure=figure, cmap=cmap)
+        
+        aplt.Array(array=arr, plotter=plotter)
+        
+        The `Plotter` object is detailed in the `autoarray.plot.plotter` package.
+        
+        The matplotlib wrapper objects in ths module also use configuration files to choose their default settings.
+        For example, in `autoarray.config.visualize.mat_base.Figure.ini` you will note the section:
+        
+        [figure]
+        figsize=(7, 7)
+        
+        [subplot]
+        figsize=auto
+        
+        This specifies that when a data structure (like the `Array` above) is plotted, the figsize will always be 
+        (7,7) when a single figure is plotted and it will be chosen automatically if a subplot is plotted. This
+        allows one to customize the matplotlib settings of every plot in a project.
+        """
+
+        self.for_subplot = False
+
+        self.kwargs = kwargs
+
+    @property
+    def config_dict(self):
+
+        if not self.for_subplot:
+
+            config_dict = conf.instance["visualize"][self.config_folder][
+                self.__class__.__name__

```

Comment:
Ah this is why you wanted the different classes.

Reply:
Lol